### PR TITLE
Update version range for eslint-plugin-jsdoc

### PR DESCRIPTION
**Pull request by dburles**

### CODE REVIEW

**Review by jaydenseric — CHANGES_REQUESTED** [review]
Please add a changelog entry under a `Next` heading.

File: changelog.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # eslint-config-env changelog
 
+## Next
+
+### Patch
+
+- Updated the eslint-plugin-jsdoc peer dependency to 25 - 34.
```
Comment (jaydenseric) [inline changelog.md]:
```suggestion
- Updated the [`eslint-plugin-jsdoc`](https://npm.im/eslint-plugin-jsdoc) peer dependency to `25 - 34`, via [#4](https://github.com/jaydenseric/eslint-config-env/pull/4).
```

**Review by jaydenseric — CHANGES_REQUESTED** [review]

**Review by jaydenseric — APPROVED** [review]

**Comment by jaydenseric** [comment]
Some other changes need to go along with this.

The `eslint-plugin-jsdoc` dev dependency needs to be updated:

https://github.com/jaydenseric/eslint-config-env/blob/c5b7900929cea5e77fb92aef2001bfba370820ac/package.json#L51

Which means updating the Node.js version support and GitHub Actions CI config in a semver major change. I might do this first on master, then we rebase this PR on that, then I'll merge.

**Review by jaydenseric — CHANGES_REQUESTED** [review]
https://github.com/jaydenseric/eslint-config-env/pull/4#issuecomment-839412972

When we action that, this peer dependency will need to be `^34.0.0`.

**Comment by dburles** [comment]
Anything for me to action?

**Review by jaydenseric — APPROVED** [review]

**Pull request merged**